File: UTESpac_MATLAB/siteIRGA/siteInfo.m
% site specific script to load site information

% enter the sonic height in input data table is ascending (true) or descending (false) when multiple sonic heights exist
info.ascending = false;

% enter orientation of sonics.  Sonic order: tables sorted alphabetically followed by columns sorted in descending order
info.sonicOrientation = [243, 128, 134, 139]; %UU1

% enter manufacturer of SAT.  1 for Campbell, 0 for RMYoung.  RMYoung v = Campbell u!
info.sonicManufact = [1, 1, 1, 1];  % 3 IRGASONs

% enter orientation of tower relative to sonic head
info.tower = 210; % FM 

% tower elevation
info.siteElevation = 1980; % (m)

% site slope angle
info.angle = 8.2; % mean of 20 m radius buffer

% enter expected table names.  Missing tables will be filled with NaNs to create consistency 
% when multiple output files are concatenated with getData.m
info.tableNames = {'FMDOL_20Hz', 'FMDOL_1min'}; % modified by Diane

% enter table scan frequencies corresponding to tableNames
info.tableScanFrequency = [20, 1/60];  %[Hz]

% enter number of columns in each .csv table.  Note that the number of columns in the output structure will 
% be 3 less than the number in the .csv file.  This is because the 4 column date vector is replaced with a Matlab's 
% single-column serial time.  Also, note that View Pro frequently cuts of column 1 (the year!) of the .csv file. 
info.tableNumberOfColumns = [48, 10]; % modified by Diane

% use local mean reference temperature from slow sensors if exists
info.useTrefHMP = true;
info.avgSlowFreq = 1; % averaging period to match fast to slow
info.shiftsSonHeight = [13.94, 32.18]; % needs to be exist in pairs with below
info.shiftsHMPHeight = [15, 30];

% seperately run sonic at 51.5 m but need to zRef at 4.42 m
info.shiftzRef = false; % only if run a single sonic at 51.5 m
info.zRefLowestSon = 4.42;

% SSITC quality-flagging settings
info.SSITC_subAvgMin = 5;

% Set this to your best estimate of canopy height.
info.canopyHeight = 19.3;

% Start with 0 for simplicity.
% You can later test 0.67*canopyHeight for above-canopy z-d correction.
info.displacementHeight = 0;

% Use canopy-aware sigma_w/u* ITC model for z <= canopyHeight.
info.useCanopyITC = true;

File: siteGill20250723_20250828/siteInfo.py
"""Site configuration for siteGill (converted from siteInfo.m)."""

sonicOrientation     = [36]  # degrees from true north
sonicManufact        = [2]   # 1 Gill WindmasterPro: 0 as RMYoung, 1 as IRGASON, 2 as Gill WindmasterPro
tower                = 210   # Tower orientation relative to Sonic
siteElevation        = 1980  # (m)
angle                = 8.2  # mean of 20 m radius buffer
tableNames           = ["FMDOL_10Hz", "FMDOL_1min"]  # fast and slow table names
tableScanFrequency   = [10, 1/60]  # [Hz]
tableNumberOfColumns = [13, 12]  # Note that the number of columns in the output structure will be 3 less than the number in the .csv file.  This is because the 4 column date vector is replaced with a single-column serial time.

# Single sonic at 51.5 m; reference T/RH/P to lowest sonic on full tower (4.42 m).
useTrefHMP    = True   # Use slow-response sensor (e.g. HMP, EE181) if exits
avgSlowFreq   = 1      # min; used to compute EC pressure at HMP freq
shiftzRef     = True   # True if needs to prelocate reference height
zRefLowestSon = 4.42   # Reference height
ascending     = False  # heights stored high → low in input tables

# SSITC quality flag settings
SSITC_subAvgMin   = 5     # sub-period length [min] for steady-state test
displacementHeight = 0    # zero-plane displacement height [m]
canopyHeight      = 19.3  # canopy height [m] for in-canopy ITC model
useCanopyITC      = True  # use Rannik et al. canopy σ_w/u* for z ≤ canopyHeight
